apps/demo-app/vite.config.ts

```diff
@@ -1,7 +1,7 @@
-import vue from "@vitejs/plugin-vue";
 import { defineConfig } from "vite";
+import { VITE_BASE_CONFIG } from "./node_modules/@sit-onyx/shared/vite.config.base";
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [vue()],
+  ...VITE_BASE_CONFIG,
```

Comment:
Would prefer to use Vite's mergeConfig so it is deep merged correctly:
```ts
import { mergeConfig, UserConfig } from "vite";
import { VITE_BASE_CONFIG } from "./node_modules/@sit-onyx/shared/vite.config.base";

// https://vitejs.dev/config/
export default mergeConfig(VITE_BASE_CONFIG, {} satisfies UserConfig);
```

Reply:
oh, interesting didn't know about that, thanks!

Reply:
@larsrickert I dropped the `mergeConfig` approach again, as it causes issues (it doesn't merge exisiting plugins) and currently solves none. The spread operator behaves exactly as expected :)

Reply:
I fear that this approach will cause unexpected issues in the future if we add/override custom options which might override ones from VITE_BASE_CONFIG because they are not merged correctly :/

What if we implement a `createViteConfig({})` utility method where overrides can be passed which uses `mergeConfig` from vite under the hood plus takes care of also merging the plugins?

Reply:
We can do that when needed :) Would prefer to keep it simple as it is right now